### ISSUE TITLE
BugFix #196:  Uncaught Error: Unexpected ] at 0 - [merged]

### ISSUE THREAD

**Issue by benloh** [theRAPTLab/gsgo#543]
In GitLab by @daveseah on May 27, 2021, 07:49

_Merges hotfix/196-parsing-bug -> dev_

There was a problem parsing ScriptText with "chained blocks" of this form:
```
ifExpr {{ false }} [[
  dbgOut 'true'
]] [[
  dbgOut 'false'
]]
```
and also this form, which nests the above inside the top-level containing block for `when`:
```
when A touches B [[
  ifExpr {{ false }} [[
    dbgOut 'true'
  ]] [[
    dbgOut 'false'
  ]]
]]
```
The root cause is that the `[[ ]] [[ ]]` needed to be parsed into **two separate blocks** instead of dumped line-by-line into a single `{ block }` token. However, this parsing only happens at the top level; for nested blocks their processing is deferred because the transpiler works recursively. For this case, the script lines themselves must be copied into a single `{block}`. 

The major changes are in `gobbleMultiblock()` in `class-gscript-tokenizer`. The `test-script-parser` unit test has also been heavily modified to perform a variety of unit tests on the different block forms.

## TESTING

I've tested the following cases, but we need to make sure all scripts are working as expected.

* [X] - Does `main?model=aquatic` work and run the simulation
* [X] - Does `DevCompiler` work when saving the blueprint?
* [ ] - Other scripts??

**NOTE** - When enabling the unit tests for compiler, etc (see `SystemInit.jsx`, `Home.jsx`, and `DevCompiler.jsx`), several **new keywords cause the unit tests to fail** because they are referring to system features that need to be available at runtime. These appear to be caused by hacks for the Spring pilot; they  need to be removed before we can move the simulator to the server. You can see what's causing the problems by uncommenting all the tests in `SystemInit.jsx` or `Home.jsx` and seeing the crashes in the Javascript console.
